report/3.Theory/SyntaxTheory.tex

```diff
@@ -0,0 +1,178 @@
+This subset, referred to as $HCL_{subset}$ can be described as the set:
+\begin{center}
+	$HCL_{subset} = \{a^nbc^n | a \in \{(\} \wedge b \in P \wedge c \in \{)\}\}$
+\end{center}
+
+where P is an arbitrary set containing elements legal in the context.
+
+To prove that HCL is not a regular language the pumping lemma will be utilized.
+The pumping lemma states that:
+\begin{center}
+	If A is a regular language, there exists a number $p>0$ so all strings $s \in A $, where $|s| \geq p$ can be divided $s = xyz$ so the following properties withhold:
+	\begin{itemize}
+		\item $xy^iz \in A$ for $i \geq 0$
+		\item $|y| > 0$
+		\item $|xy| leq p$
+	\end{itemize}
+\end{center}
+
+First the division is chosen for s.
+$s = xyz$, where $s = (((...b)))...$.
+Meaning that x consists only of left parentheses, y consists of a concatenation of left parentheses and elements from P and finally z consists only of right parentheses.
+
+The length of the string is $p < |s| < (p3)/2$, since $s = a^nbc^n$, where $n = p/2$ and $0 < |b| \leq p/2$.
+
+This means that the third condition is upheld since $|xy|$ cannot be greater than p.
+The second condition is also upheld since y consist of at least some left parentheses. 
+If the string does contain right parentheses, which it does. 
+Y cannot be of the length 0.
+
+The first condition is not upheld since if $i = 0$ the new string cannot contain an equal amount of left and right parentheses because of the same reason as to why the second condition is upheld.
+
+This concludes the proof by contradiction.
+Since HCl does not uphold the required property for it to be a regular language, it it not and the only automaton capable of recognizing the language is the NDPA described in the previous sections.
+
+\subsection{Types and Identifiers}
+So far the syntax of HCL has been properly defined as a CFL. 
+Meaning that all Variables, terminals and the transition rules for recognizing HCl has been described in a computational model as seen on page 
+%Indsæt sidetal for HCL CFG definition.
+and in appendix.
+%Indæst reference til appendix hvor NPDA for HCL er beskrevet.
+It is now relevant to define how the types and identifiers in HCL can be described or recognized by a computational model.
+The goal is to formally describe the languages $L_{none}$, $L_{bool}$, $L_{number}$, $L_{text}$ which are the type languages and the identifier language $L_{identifiers}$.
+
+An alphabet is a finite set of symbols, a string is a finite sequence of symbols and a language is a set of strings.
+To describe the languages mentioned above, the Alphabets used in the informal and formal notations needs to be defined. 
+These can be seen in Table 3.1.
+
+\begin{table}[!htb]
+	\centering
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Alphabet} & \textbf{Set}                            \\ \hline
+		T                 & \{true\}                                \\ \hline
```

Comment:
True - not true
False - not false

Reply:
fixed

Reply:
Tror at casper mener, at vi skriver med stort forbogstav til boolean literals.